### DISCUSSION TITLE
Episode 230 Stack vs Heap - Thread safety?

### DISCUSSION THREAD

**Discussion by grennis** [pointfreeco/episode-code-samples#139] [Q&A]
I was wondering if the Wrapper class is thread safe. It seems like the check for `isKnownUniquelyReferenced` and the code that acts on the result should be locked because another thread could create another reference in between those statements. This could affect the performance test. 

I'm not sure this is actually an issue though?

**Comment by stephencelis** [comment]
@grennis There's a good thread on the thread safety of `isKnownUniquelyReferenced` here, particularly John McCall's reply: https://forums.swift.org/t/isknownuniquelyreferenced-thread-safety/22933/5

In short, we don't need to manually lock here because the `inout` nature of the input has a precondition against allowing concurrent writes.
